File: gluecodium/src/test/resources/smoke/typedefs/output/android/com/example/smoke/TypeDefs.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.List;
public final class TypeDefs extends NativeBase {
    public static final class StructHavingAliasFieldDefinedBelow {
        public double field;
        public StructHavingAliasFieldDefinedBelow(final double field) {
            this.field = field;
        }
    }
    public static final class TestStruct {
        @NonNull
        public String something;
        public TestStruct(@NonNull final String something) {
            this.something = something;
        }
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected TypeDefs(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    public static native double methodWithPrimitiveTypeDef(final double input);
    @NonNull
    public static native List<TypeDefs.TestStruct> methodWithComplexTypeDef(@NonNull final List<TypeDefs.TestStruct> input);
    public static native double returnNestedIntTypeDef(final double input);
    @NonNull
    public static native TypeDefs.TestStruct returnTestStructTypeDef(@NonNull final TypeDefs.TestStruct input);
    @NonNull
    public static native TypeDefs.TestStruct returnNestedStructTypeDef(@NonNull final TypeDefs.TestStruct input);
    @NonNull
    public static native TypeCollection.Point returnTypeDefPointFromTypeCollection(@NonNull final TypeCollection.Point input);
    @NonNull
    public native List<Double> getPrimitiveTypeProperty();
    public native void setPrimitiveTypeProperty(@NonNull final List<Double> value);
}

File: gluecodium/src/test/resources/smoke/typedefs/output/android-kotlin/com/example/smoke/TypeDefs.kt
/*

 *
 */

@file:JvmName("TypeDefsExtensions")

package com.example.smoke

import com.example.NativeBase

class TypeDefs : NativeBase {

    class StructHavingAliasFieldDefinedBelow {
        @JvmField var field: Double



        constructor(field: Double) {
            this.field = field
        }





    }

    class TestStruct {
        @JvmField var something: String



        constructor(something: String) {
            this.something = something
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    var primitiveTypeProperty: MutableList<Double>
        external get
        external set




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)


        @JvmStatic external fun methodWithPrimitiveTypeDef(input: Double) : Double


        @JvmStatic external fun methodWithComplexTypeDef(input: MutableList<TypeDefs.TestStruct>) : MutableList<TypeDefs.TestStruct>


        @JvmStatic external fun returnNestedIntTypeDef(input: Double) : Double


        @JvmStatic external fun returnTestStructTypeDef(input: TypeDefs.TestStruct) : TypeDefs.TestStruct


        @JvmStatic external fun returnNestedStructTypeDef(input: TypeDefs.TestStruct) : TypeDefs.TestStruct


        @JvmStatic external fun returnTypeDefPointFromTypeCollection(input: TypeCollection.Point) : TypeCollection.Point
    }
}

